core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3219,6 +3227,34 @@ public void validateIntervalQualifier(SqlIntervalQualifier qualifier) {
     }
   }
 
+  @Override public void validateCreateTable(SqlCreateTable createTable) {
+    Preconditions.checkArgument(createTable != null);
+    // Type doesn't matter for CREATE TABLE statements.
+    nodeToTypeMap.put(createTable, unknownType);
+    CalciteSchema schema = catalogReader.getRootSchema();
+    RelDataTypeFactory.Builder builder = typeFactory.builder();
+    if (createTable.columnList == null) {
```

Comment:
Is it legal for the column list be null (not just empty, but null)? According to the Dialect1 docs, it doesn't look legal and I'm not sure of a dialect in which it would be.

Reply:
I'm also unsure, I went off of the code of SqlCreateTable where it states that the column list "may be null", WDYT? https://github.com/googleinterns/calcite/blob/67b7bb66aabf05a58171096356dc71476bf76381/core/src/main/java/org/apache/calcite/sql/SqlCreateTable.java#L88. It also defaults to null if there is no column list here: https://github.com/googleinterns/calcite/blob/67b7bb66aabf05a58171096356dc71476bf76381/parsing/dialects/intermediate/dialects/dialect1/parserImpls.ftl#L936